### DISCUSSION TITLE
Using a VCL form from an old Delphi project in a Python project

### DISCUSSION THREAD

**Discussion by sbhoek** [Embarcadero/DelphiVCL4Python#88] [Q&A]
Hi! I used to develop programmes in Delphi, but these days we are supposed to develop in Python - often prototypes. Serious stuff is done in Java. I'm developing in Python anyway. Until not too long ago we used to maintain a number of applications realised in Delphi. In almost each case Delphi was used to provide an interface to a specialised DLL. Those applications are now considered legacy and new versions may have to be developed more or less from scratch. I'm thinking of reusing at least the graphical interfaces. I have just tried delphivcl for the first time. Thanks for this package! Unfortunately, I have only seen examples for delphifmx so far. Anyway, I have managed to load a very simple Delphi form called Unit2.dfm whilst using python 3.9. An error appeared though: <built-in method LoadProps of TUnit2 object at 0x0000016DC01F08B0> returned a result with an error set. Maybe I did something wrong? How can I retrieve more details about the errors? Should I use a different Python version? Any help will be appreciated!

**Comment by sbhoek** [comment]
Hmm, no comments so far. Maybe I should be more clear. This is part of my code. I read about exporting TForm descendants. Is it necessary? Why can we not load them directly?

from pathlib import Path
from delphivcl import Form

class TUnit2(Form):
    
    def __init__(self, owner):
        try:
            curdir = Path(__file__).parent.absolute()
            self.LoadProps(curdir / "Unit2.dfm")
            pass
        except Exception as e:
            print(e)

**Reply by lmbelo** [reply]
Hello @sbhoek,

Can you share a sample project, please? Only loading a "dfm" shouldn't fail, probably there's something else in your scripts.

**Reply by lmbelo** [reply]
Sorry for the delay, I didn't see your post before.

**Reply by sbhoek** [reply]
[DelphiVcl.zip](https://github.com/Embarcadero/DelphiVCL4Python/files/15016170/DelphiVcl.zip)
Dear Lucas, please find my project attached in zipped form. There's no hurry in life ;-)
